README.md

```diff
@@ -1 +1,44 @@
-# MultiDID
+# multidid
+
+> A very compact representation of DIDs and DID URLs
+
+Multidid is a representation strategy for DIDs and DID URLs that is very compact and extensible. It allows any DID method to be represented as a string of bytes.
+
+## Format
+
```

Comment:
I think we should just use IPLD schema extension syntax to define multiformats, otherwise they're just too vague I'd suggest something like

```ipldsch
type MultiDID union {
  | DID   "0x0d1d"
} representation varintprefix

type DID struct {
  # DID method-name + method-specific-id
  id                         Method

  # DID URL path, query & fragment are length prefixed bytes
  location               Location
} representation varintbytes

type Method union {
   # did:key
   Key          "0xe7"
   Key          "0xeb"
   Key          "0xec"
   Key          "0xed"
   Key          "0x1200"
   Key          "0x1201"
   Key          "0x1202"
   Key          "0x1205"

   # did:pkh
   PKH         "0xca"

   # did:*
   Raw         "0x55"
} representation varintprefix

type Location bytes representation lengthprefixed
```

That way we can go ahead and define `Raw` `Key` and `PKH` as separate schemas like this.

Reply:
How about the same way we define varsig?

```xml
<varint 0x0d1d><varint method_code><method_id_bytes><varint url_length><url_bytes>
```

I worry that using new words like `varintprefix` and `lengthprefixed` makes it harder for people who are not familiar with multiformats because they are not aware of how these layouts usually look like.

Reply:
> How about in a similar way as we define varsig?

I was in fact going to suggest the same thing in varsig spec. I would much rather write somewhere what we mean by `varintprefix` and `lengthprefixed` than use syntax that has no definition at all.

That said those are just suggestions, it's up to you what makes most sense.

Ideally we'd even have visuals like ssb and dat protocols have

https://ssbc.github.io/scuttlebutt-protocol-guide/#rpc-protocol
https://dat-ecosystem-archive.github.io/how-dat-works/#mdns-response

And I was hoping we could generate such things from the extended schema